packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -695,13 +666,14 @@ export const DatePickerMixin = (subclass) =>
     _selectDate(dateToSelect) {
       const value = this._formatISO(dateToSelect);
 
-      // Only set flag if the value will change.
-      if (this.value !== value) {
-        this.dirty = true;
-        this.__dispatchChange = true;
-      }
+      const shouldDispatchChange = this.value !== value;
 
       this._selectedDate = dateToSelect;
+
+      if (shouldDispatchChange) {
```

Comment:
**note:** This ensures that the `change` event is dispatched after `value-changed` and thus allows removing `_propertiesChanged` and `_updated` above.